pillar/vault/roles/pki.sls

```diff
@@ -0,0 +1,46 @@
+{% set env_settings = salt.cp.get_file_str("salt://environment_settings.yml")|load_yaml %}
+{% set ttl = '8760h' %}
+{% set ou = 'Open Learning' %}
+{% set org = 'Massachusetts Institute of Technology' %}
+{% set country = 'US' %}
+{% set locality = 'Cambridge' %}
+{% set street_address = '77 Massachusetts Ave' %}
+{% set postal_code = '02139' %}
+
+vault:
+  roles:
+    {% for env in env_settings.environments %}
+    {% for app in env.backends.pki %}
+    {{ app }}-{{ env }}-pki:
+      backend: pki_int_{{ env }}
+      {% for type in ['client', 'server'] %}
+      name: pki-{{ env }}-{{ app }}-{{ type }}
+      {% endfor %}
+      ttl: {{ ttl }}
+      max_ttl: {{ ttl }}
+      allowed_domains:
+        - {{ app }}.service.consul
+        - nearest-{{ app }}.service.consul
```

Comment:
For the `nearest` services it's a stored query so the domain ends up being `.query.consul` rather than `.service.consul`